de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/NewFileHandler.java

```diff
@@ -47,13 +31,8 @@ public void execute(IEclipseContext context, RCPDialogService dialogService, RCP
 			directory = FileHandlingUtility.getPath(fileSystem.getWorkspaceDirectory());
 		}
 
-		if (context != null) {
-			this.context = context;
-		}
-
-		NewFileWizard newFileWizard = new NewFileWizard(this.context, directory);
-		WizardDialog dialog = new WizardDialog(shell, newFileWizard);
-		dialog.open();
+		NewFileDialog dialog = new NewFileDialog(directory, imageService);
+		dialog.showDialog();
 	}
 
 	@CanExecute

```

Comment:
Inconsistent to every other handler.

Reply:
changed to dialog.open()

Reply:
I meant the expression contained in canExecute is inconsistent to the canExecute of other handlers (namely new folder) when it should be the same (referencing the method below)

Reply:
canExecute methods were unified